src/back/spv/writer.rs

```diff
@@ -1610,24 +1635,66 @@ impl Writer {
         }
     }
 
+    fn create_block(&mut self) -> Block {
+        let mut block = Block::default();
+        let label_id = self.generate_id();
+        let label_instruction = super::instructions::instruction_label(label_id);
+        block.label_id = Some(label_id);
+        block.label = Some(label_instruction);
+        block
+    }
+
     fn write_block(
         &mut self,
-        statements: &[crate::Statement],
+        statements: &crate::Block,
         ir_module: &crate::Module,
         ir_function: &crate::Function,
         function: &mut Function,
-    ) -> Result<spirv::Word, Error> {
-        let mut block = Block::default();
-        let id = self.generate_id();
-        block.label = Some(super::instructions::instruction_label(id));
+        branch_id: Option<spirv::Word>,
+    ) -> Result<Block, Error> {
+        let mut block = self.create_block();
 
         for statement in statements {
             match *statement {
                 crate::Statement::Block(ref ir_block) => {
-                    if !ir_block.is_empty() {
-                        //TODO: link the block with `OpBranch`
-                        self.write_block(ir_block, ir_module, ir_function, function)?;
-                    }
+                    // TODO
+                }
+                crate::Statement::If {
+                    ref condition,
+                    ref accept,
+                    ref reject,
+                } => {
+                    let (condition_id, _) = self.write_expression(
+                        ir_module,
+                        ir_function,
+                        *condition,
+                        &mut block,
+                        function,
+                    )?;
+
+                    let pdom_label = self.generate_id();
+
+                    block
+                        .body
+                        .push(super::instructions::instruction_selection_merge(
+                            pdom_label,
+                            spirv::SelectionControl::NONE,
+                        ));
+
+                    let accept_block = self.write_block(accept, ir_module, ir_function, function, Some(pdom_label))?;
+                    let reject_block = self.write_block(reject, ir_module, ir_function, function, Some(pdom_label))?;
+
+                    block.termination = Some(super::instructions::instruction_branch_conditional(
+                        condition_id,
+                        accept_block.label_id.unwrap(),
+                        reject_block.label_id.unwrap(),
+                    ));
+                    function.blocks.push(block);
+                    function.blocks.push(accept_block);
+                    function.blocks.push(reject_block);
+
+                    let pdom_block = self.create_block();
+                    block = pdom_block;
```

Comment:
Remaining issues are: 
- How do we add the `pdom_block` to the `function.blocks` vector?
- How do we add the main block to the `function.block` vector when not having branching statements?